npm login w/ custom cert + key is broken in npm@5.5

#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [x] Other (_see below for feature requests_):

#### What's going wrong?
`npm login` doesn't use the provided cert and key, causing `EPROTO` errors on custom registry with mutual TLS.

Works in `npm@5.3` and `npm@5.4`. In 5.5 I get:
```
npm ERR! code EPROTO
npm ERR! errno EPROTO
npm ERR! request to https://${registry}/-/user/org.couchdb.user:${myuser} failed, reason: write EPROTO 140737250317248:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:../deps/openssl/openssl/ssl/s3_pkt.c:1494:SSL alert number 40
npm ERR! 140737250317248:error:1409E0E5:SSL routines:ssl3_write_bytes:ssl handshake failure:../deps/openssl/openssl/ssl/s3_pkt.c:659
```

Other npm requests work fine w/ the certs, just login is failing.

#### How can the CLI team reproduce the problem?
```
; ~/.npmrc
cert = "..."
key = "..."
```

`npm login`

### supporting information:
```
$ npm -v
5.5.1
$ node -v
8.9.0
```

--- a/lib/auth/legacy.js
+++ b/lib/auth/legacy.js
@@ -4,6 +4,8 @@ const profile = require('npm-profile')
 const log = require('npmlog')
 const npm = require('../npm.js')
 const output = require('../utils/output.js')
+const pacoteOpts = require('../config/pacote')
+const fetchOpts = require('../config/fetch-opts')
 
 module.exports.login = function login (creds, registry, scope, cb) {
   let username = creds.username || ''
@@ -22,7 +24,10 @@ module.exports.login = function login (creds, registry, scope, cb) {
     email = e
     return profile.login(username, password, {registry: registry, auth: auth}).catch((err) => {
       if (err.code === 'EOTP') throw err
-      return profile.adduser(username, email, password, {registry: registry})
+      return profile.adduser(username, email, password, {
+        registry: registry,
+        opts: fetchOpts.fromPacote(pacoteOpts())
+      })
     }).catch((err) => {
       if (err.code === 'EOTP' && !auth.otp) {
         return read.otp('Authenicator provided OTP:').then((otp) => {

--- a/lib/config/fetch-opts.js
+++ b/lib/config/fetch-opts.js
@@ -1,0 +1,77 @@
+'use strict';
+
+const url = require('url')
+
+module.exports.fromPacote = fromPacote
+
+function fromPacote (opts) {
+  return {
+    cache: getCacheMode(opts),
+    cacheManager: opts.cache,
+    ca: opts.ca,
+    cert: opts.cert,
+    headers: getHeaders('', opts.registry, opts),
+    key: opts.key,
+    localAddress: opts.localAddress,
+    maxSockets: opts.maxSockets,
+    proxy: opts.proxy,
+    referer: opts.refer,
+    retry: opts.retry,
+    strictSSL: !!opts.strictSSL,
+    timeout: opts.timeout,
+    uid: opts.uid,
+    gid: opts.gid
+  }
+}
+
+function getCacheMode (opts) {
+  return opts.offline
+  ? 'only-if-cached'
+  : opts.preferOffline
+  ? 'force-cache'
+  : opts.preferOnline
+  ? 'no-cache'
+  : 'default'
+}
+
+function getHeaders (uri, registry, opts) {
+  const headers = Object.assign({
+    'npm-in-ci': opts.isFromCI,
+    'npm-scope': opts.projectScope,
+    'npm-session': opts.npmSession,
+    'user-agent': opts.userAgent,
+    'referer': opts.refer
+  }, opts.headers)
+  // check for auth settings specific to this registry
+  let auth = (
+    opts.auth &&
+    opts.auth[registryKey(registry)]
+  ) || opts.auth
+  // If a tarball is hosted on a different place than the manifest, only send
+  // credentials on `alwaysAuth`
+  const shouldAuth = auth && (
+    auth.alwaysAuth ||
+    url.parse(uri).host === url.parse(registry).host
+  )
+  if (shouldAuth && auth.token) {
+    headers.authorization = `Bearer ${auth.token}`
+  } else if (shouldAuth && auth.username && auth.password) {
+    const encoded = Buffer.from(
+      `${auth.username}:${auth.password}`, 'utf8'
+    ).toString('base64')
+    headers.authorization = `Basic ${encoded}`
+  } else if (shouldAuth && auth._auth) {
+    headers.authorization = `Basic ${auth._auth}`
+  }
+  return headers
+}
+
+function registryKey (registry) {
+  const parsed = url.parse(registry)
+  const formatted = url.format({
+    host: parsed.host,
+    pathname: parsed.pathname,
+    slashes: parsed.slashes
+  })
+  return url.resolve(formatted, '.')
+}

--- a/lib/config/fetch-opts.js
+++ b/lib/config/fetch-opts.js
@@ -1,4 +1,4 @@
-'use strict';
+'use strict'
 
 const url = require('url')
 

